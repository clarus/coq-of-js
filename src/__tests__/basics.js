// @flow
import {compileAndPrint} from "../compiler/index.js";

it("handles basics", () => {
  expect(
    compileAndPrint(
      `// Some examples

const
  b: boolean = false && true,
  n: number = -12 + 23;

const s = "hi";

const a = [1, (2 : number), 3];

const cond = b ? "a" : 'b';

function id<A, B>(x: A): A {
  return x;
}

function basicTypes(n: number, m: number): string {
  return "OK";
}

const r = id(basicTypes(12, 23));

const f = function<A> (x : A, y : A): bool {
  return true;
}

const arrow = x => x + 1;
`,
      40,
    ),
  ).toMatchSnapshot();
});
