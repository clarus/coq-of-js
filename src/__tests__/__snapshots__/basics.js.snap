// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles basics 1`] = `
"Module Rec.
  Record t := {
    a : string;
    b : Z;
    c : bool;
  }.
End Rec.
Definition Rec := Rec.t.

Definition o :=
  {|
    Rec.a := \\"hi\\";
    Rec.b := 12;
    Rec.c := false;
  |}.

Module Status.
  Module Error.
    Record t := {
      message : string;
    }.
  End Error.

  Inductive t :=
  | Error (_ : Error.t)
  | Loading
  | Nothing.
End Status.
Definition Status := Status.t.

Definition status : Status :=
  Status.Error
  {| Status.Error.message := \\"hi\\"; |}.

Definition b : bool :=
  false && true.

Definition n : Z :=
  (- 12) + 23.

Definition s :=
  \\"hi\\".

Definition a :=
  [ 1, (2 : Z), 3 ].

Definition cond :=
  if b then \\"a\\" else \\"b\\".

Definition id
  {A B : Type}
  (x : A)
  : A :=
  x.

Definition basicTypes
  (n : Z)
  (m : Z)
  : string :=
  \\"OK\\".

Definition r :=
  id (basicTypes 12 23).

Definition f :=
  fun {A : Type} (x : A) (y : A) =>
    true.

Definition arrow :=
  fun x => x + 1."
`;
